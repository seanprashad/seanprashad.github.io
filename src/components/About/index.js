import React from 'react';
import { Container, Col, Row } from 'reactstrap';
import ReactTooltip from 'react-tooltip';
import { Event } from '../Shared/Tracking';

import aboutImg from '../../images/about.png';

import './styles.scss';

const About = () => {
  return (
    <Container className="about" fluid>
      <ReactTooltip place="bottom" />
      <Col>
        <Row>
          <h2>About Me</h2>
        </Row>
        <Col>
          <Row>
            <Col sm={2}>
              <img src={aboutImg} alt="profile" />
            </Col>
          </Row>
          <Row>
            <Col sm={7}>
              <Row>
                <Row>
                  <p>Hey! I&apos;m Sean, a Software Engineer at Square.</p>
                </Row>
                <Row>
                  <p>
                    In my spare time, you can find me{' '}
                    <a
                      href="https://bit.ly/instagram-seanprashad"
                      target="instagram"
                      onClick={() =>
                        Event(
                          'About Me',
                          'Clicked Instagram link',
                          'Instagram link',
                        )
                      }
                    >
                      <strong>training</strong>
                    </a>
                    , swimming, cooking, or doing handy work around the house.
                  </p>
                </Row>
              </Row>
            </Col>
          </Row>
        </Col>
      </Col>
    </Container>
  );
};

export default About;
