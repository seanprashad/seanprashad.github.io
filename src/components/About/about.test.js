import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import About from '.';

it('matches the snapshot', () => {
  const tree = renderer.create(<About />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the About component without crashing', () => {
  shallow(<About />);
});

it('renders the main header', () => {
  const wrapper = shallow(<About />);

  expect(wrapper.find('h2').html()).toEqual('<h2>About Me</h2>');
});

it('renders the headshot image', () => {
  const wrapper = shallow(<About />);

  expect(wrapper.find('img').html()).toEqual(
    '<img src="about.png" alt="profile"/>',
  );
});

it('renders the subheader', () => {
  const wrapper = shallow(<About />);

  expect(wrapper.find('p')).toHaveLength(2);

  expect(
    wrapper
      .find('p')
      .at(0)
      .html(),
  ).toEqual('<p>Hey! I&#x27;m Sean, a Software Engineer at Square.</p>');

  expect(
    wrapper
      .find('p')
      .at(1)
      .html(),
  ).toEqual(
    '<p>In my spare time, you can find me training, swimming, cooking, gardening, or doing handy work around the house.</p>',
  );
});
