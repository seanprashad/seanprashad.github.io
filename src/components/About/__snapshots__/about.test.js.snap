// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="about container-fluid"
>
  <div
    className="__react_component_tooltip place-bottom type-dark"
    data-id="tooltip"
  />
  <div
    className="col"
  >
    <div
      className="row"
    >
      <h2>
        About Me
      </h2>
    </div>
    <div
      className="col"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-2"
        >
          <img
            alt="profile"
            src="about.png"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-6"
        >
          <div
            className="row"
          >
            <div
              className="row"
            >
              <p>
                Hey! I'm Sean, a Software Engineer at Square.
              </p>
            </div>
            <div
              className="row"
            >
              <p>
                In my spare time, you can find me
                 
                <a
                  href="https://bit.ly/instagram-seanprashad"
                  onClick={[Function]}
                  target="instagram"
                >
                  <strong>
                    training
                  </strong>
                </a>
                , swimming, cooking, or doing handy work around the house.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
